packages/autorest.python/autorest/codegen/serializers/builder_serializer.py

```diff
@@ -815,72 +810,75 @@ def _initialize_overloads(
+            overload_retval: List[str] = []
+            client_names = [
+                overload.request_builder.parameters.body_parameter.client_name
+                for overload in builder.overloads
+            ]
+            for v in sorted(set(client_names), key=client_names.index):
+                overload_retval.append(f"_{v}: Any = None")
+
+            # make sure some special type is in last position and some in first position
+            # but we can't change original data
+            overloads_copy = [
+                o
+                for o in builder.overloads
+                if o.parameters.body_parameter.type.enable_overload_check
+            ]
+            for i, _ in enumerate(overloads_copy):
+                if isinstance(
+                    overloads_copy[i].parameters.body_parameter.type,
+                    (AnyType, AnyObjectType),
+                ):
+                    _swap(overloads_copy, i, -1)
+                if isinstance(
+                    overloads_copy[i].parameters.body_parameter.type, BinaryType
+                ):
+                    _swap(overloads_copy, i, 0)
+
+            if_statement = "if"
+            for idx, overload in enumerate(overloads_copy):
                 body_param = overload.parameters.body_parameter
-                retval.append(
-                    f"{if_statement} {body_param.type.instance_check_template.format(body_param.client_name)}:"
-                )
-                if body_param.default_content_type and not same_content_type:
-                    retval.append(
-                        f'    content_type = content_type or "{body_param.default_content_type}"'
+                try:
+                    if idx + 1 >= len(overloads_copy):
+                        raise ValueError()
+                    overload_retval.append(
+                        f"{if_statement} {body_param.type.instance_check_template.format(body_param.client_name)}:"
                     )
-                retval.extend(
+                except ValueError:
+                    overload_retval.append("else:")
+                overload_retval.extend(
                     f"    {l}"
                     for l in self._create_body_parameter(cast(OperationType, overload))
                 )
+                if body_param.default_content_type is None:
+                    overload_retval.extend(
+                        [
+                            f"    {l}"
+                            for l in _content_type_check(body_param.content_types)
+                        ]
+                    )
+                elif not same_content_type:
+                    overload_retval.append(
+                        f'    content_type = content_type or "{body_param.default_content_type}"'
+                    )
+                if_statement = "elif"
+            retval.extend(overload_retval)
+        except Exception:  # pylint: disable=broad-except
```

Comment:
what exception are we trying to catch here, what could go wrong?

Reply:
udpated